spec/player_choice_spec.rb

```diff
@@ -1,10 +1,24 @@
 require 'rspec'
 require './lib/player_choice'
 require './lib/board'
+require 'stringio'
+
+class TestUI
+  attr_reader :sorry_taken_called, :sorry_invalid_move
+
+  def sorry_taken
+    @sorry_taken_called = true
+  end
+
+  def sorry_invalid
+    @sorry_invalid_move = true
+  end
+end
 
 describe 'PlayerChoice' do
   before(:each) do
-    @choice = PlayerChoice.new
+    @test_ui = TestUI.new
+    @choice = PlayerChoice.new(output: @test_ui)
```

Comment:
Great job injecting the `TestUI` class in order to test output without cluttering test runs with `put` statements!